lib/events/filelog.go

```diff
@@ -115,6 +115,9 @@ type FileLog struct {
 
 // EmitAuditEvent adds a new event to the log.
 func (l *FileLog) EmitAuditEvent(ctx context.Context, event AuditEvent) error {
+	l.mu.Lock()
```

Comment:
@fspmarshall Doesn't this change basically cause I/O to be done under lock? And rotateLog too, not sure what it actually does and how grabbing a lock over it will affect performance. Wouldn't it make sense to just make a getter that returns `l.file` under lock or something?

Reply:
@r0mant The file object in question gets closed and replaced at regular intervals by the rotation logic, so a simple getter wouldn't be sufficient.  That being said, I can probably improve this with optimistic RW locking.  I'll give that a shot.